src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
     public static void compareResult(Map<String, Object> optionsMap) throws Exception {
         String benchmarkResultPrevious = optionsMap.get("--benchmark-result-previous").toString();
         String benchmarkResultNew = optionsMap.get("--benchmark-result-new").toString();
-        String outputFilePath = optionsMap.get("<output_file>").toString();
-        Map<String, BigDecimal> scoreMap = new HashMap<>();
+        Map<String, Double> comparisonResults = new HashMap<>();
         for (String benchmarkScoreKeyword : BENCHMARK_SCORE_KEYWORDS) {
-            BigDecimal previousScore = getScore(benchmarkResultPrevious, benchmarkScoreKeyword);
-            BigDecimal newScore = getScore(benchmarkResultNew, benchmarkScoreKeyword);
-            BigDecimal result = calculateDifference(previousScore, newScore);
-            scoreMap.put(benchmarkScoreKeyword, result);
+            double[] previousData = preProcess(benchmarkResultPrevious, benchmarkScoreKeyword);
+            double[] newData = preProcess(benchmarkResultNew, benchmarkScoreKeyword);
+            double comparisonResult = detectRegression(previousData, newData);
+            if (comparisonResult > 0) {
```

Comment:
Replying to this [comment](https://github.com/amazon-ion/ion-java-benchmark-cli/pull/58/files/2b02d3a1cd79441eff09db1aa3a2ba4c23cc460e#r1355682971) 
`detectRegression()` method will return a non-zero value both for a significant regression and a significant improvement and only significant regression (when comparison result > 0) will be added to the final `comparisonResults` map.